lib/src/embedded/isolate_dispatcher.dart

```diff
@@ -188,7 +196,11 @@ class IsolateDispatcher {
       compilationId ?? errorId,
       handleError(error, stackTrace, messageId: messageId),
     );
-    _channel.sink.close();
+    if (_gracefulShutdown) {
+      _channel.sink.close();
+    } else {
+      exit(exitCode);
+    }
```

Comment:
This is also changing the current behavior for the Dart VM.

Reply:
Actually it's not changing any behavior for end users, see https://github.com/sass/dart-sass/pull/2413#discussion_r2023894593